test/Dialect/FIRRTL/lower-types.mlir

```diff
@@ -973,3 +973,449 @@ module  {
+// CHECK:   %c1_ui2_3 = firrtl.constant 1 : !firrtl.uint<2>
+// CHECK:   %17 = firrtl.eq %sel, %c1_ui2_3 : (!firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<1>
+// CHECK:   %18 = firrtl.and %16, %17 : (!firrtl.uint<1>, !firrtl.uint<1>) -> !firrtl.uint<1>
+// CHECK:   %19 = firrtl.mux(%18, %a_1_0, %14) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+// CHECK:   %c1_ui2_4 = firrtl.constant 1 : !firrtl.uint<2>
+// CHECK:   %20 = firrtl.eq %sel, %c1_ui2_4 : (!firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<1>
+// CHECK:   %21 = firrtl.and %c1_ui1, %20 : (!firrtl.uint<1>, !firrtl.uint<1>) -> !firrtl.uint<1>
+// CHECK:   %c1_ui2_5 = firrtl.constant 1 : !firrtl.uint<2>
+// CHECK:   %22 = firrtl.eq %sel, %c1_ui2_5 : (!firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<1>
+// CHECK:   %23 = firrtl.and %21, %22 : (!firrtl.uint<1>, !firrtl.uint<1>) -> !firrtl.uint<1>
+// CHECK:   %24 = firrtl.mux(%23, %a_1_1, %19) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+// CHECK:   %c2_ui2_6 = firrtl.constant 2 : !firrtl.uint<2>
+// CHECK:   %25 = firrtl.eq %sel, %c2_ui2_6 : (!firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<1>
+// CHECK:   %26 = firrtl.and %c1_ui1, %25 : (!firrtl.uint<1>, !firrtl.uint<1>) -> !firrtl.uint<1>
+// CHECK:   %c1_ui2_7 = firrtl.constant 1 : !firrtl.uint<2>
+// CHECK:   %27 = firrtl.eq %sel, %c1_ui2_7 : (!firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<1>
+// CHECK:   %28 = firrtl.and %26, %27 : (!firrtl.uint<1>, !firrtl.uint<1>) -> !firrtl.uint<1>
+// CHECK:   %29 = firrtl.mux(%28, %a_1_2, %24) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+// CHECK:   %c3_ui2_8 = firrtl.constant 3 : !firrtl.uint<2>
+// CHECK:   %30 = firrtl.eq %sel, %c3_ui2_8 : (!firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<1>
+// CHECK:   %31 = firrtl.and %c1_ui1, %30 : (!firrtl.uint<1>, !firrtl.uint<1>) -> !firrtl.uint<1>
+// CHECK:   %c1_ui2_9 = firrtl.constant 1 : !firrtl.uint<2>
+// CHECK:   %32 = firrtl.eq %sel, %c1_ui2_9 : (!firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<1>
+// CHECK:   %33 = firrtl.and %31, %32 : (!firrtl.uint<1>, !firrtl.uint<1>) -> !firrtl.uint<1>
+// CHECK:   %34 = firrtl.mux(%33, %a_1_3, %29) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+// CHECK:   %c0_ui2_10 = firrtl.constant 0 : !firrtl.uint<2>
+// CHECK:   %35 = firrtl.eq %sel, %c0_ui2_10 : (!firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<1>
+// CHECK:   %36 = firrtl.and %c1_ui1, %35 : (!firrtl.uint<1>, !firrtl.uint<1>) -> !firrtl.uint<1>
+// CHECK:   %c2_ui2_11 = firrtl.constant 2 : !firrtl.uint<2>
+// CHECK:   %37 = firrtl.eq %sel, %c2_ui2_11 : (!firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<1>
+// CHECK:   %38 = firrtl.and %36, %37 : (!firrtl.uint<1>, !firrtl.uint<1>) -> !firrtl.uint<1>
+// CHECK:   %39 = firrtl.mux(%38, %a_2_0, %34) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
```

Comment:
This test is too specific - it looks like you copied the output into the test and added check lines.  This will be a big maintenance burden going forward.  Please reduce the specificity here, checking the structure of the thing you want generated, instead of an exact match of the output.

Reply:
Sorry, I copy pasted the output. Will update all the lit tests to check only for the specific structure.